src/super_gradients/common/registry/__init__.py

```diff
@@ -1,4 +0,0 @@
-from super_gradients.common.registry.registry import register_model, register_metric, register_loss, register_detection_module, register_lr_scheduler
-
-
-__all__ = ["register_model", "register_detection_module", "register_metric", "register_loss", "register_lr_scheduler"]
```

Comment:
This highly contributes to the circular loop...
We might be able to keep it if we refactor EVERY register function the same way I propose in this PR (see description of PR)